data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/buffer/InMemoryBuffer.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.buffer;
+
+import java.util.ArrayList;
+
+public class InMemoryBuffer implements Buffer {
```

Comment:
I think you need flush API to flush the buffer contents to Cloud Watch, right?

Reply:
I separated the dependency of the flushing from the Buffer, the buffer is purely for holding information. The next PR will include a CloudWatchLogsClientService class which uses the SDK and Buffer to flush to CloudWatchLogs. Is this what you meant or was it some other flush API?